ocaml/testsuite/tests/typing-implicit-source-positions/recursion.ml

```diff
@@ -0,0 +1,9 @@
+(* TEST
+   * expect
+*)
+
+let rec f ~(src_pos:[%src_pos]) i = 
+  if i < 0 then 0 else f ~src_pos (i - 1)
+[%%expect {|
+val f : src_pos:[%src_pos] -> int -> int = <fun>
+|}]
```

Comment:
This test is good, but a better test would check to see whether type-based disambiguation works. Something like

```
let rec f ~(src_pos:[%src_pos]) i = f ~src_pos:{ pos_fnum = ... ; ... }
```

where `pos_fnum` is shadowed with a different field name. That is, the only reason the program above would be accepted is if the compiler used the `:[%src_pos]` to know that the argument to `f` must be a `lexing_position`.